src/forward/arweave-fetcher.go

```diff
@@ -0,0 +1,224 @@
+package forward
+
+import (
+	"github.com/warp-contracts/syncer/src/utils/config"
+	"github.com/warp-contracts/syncer/src/utils/model"
+	"github.com/warp-contracts/syncer/src/utils/monitoring"
+	"github.com/warp-contracts/syncer/src/utils/task"
+	"github.com/warp-contracts/syncer/src/utils/tool"
+	"gorm.io/gorm"
+)
+
+// Gets L1 (Arweave) interactions from the DB in batches
+// Fills in last_sort_key for each interaction before emiting to the output channel
+type ArweaveFetcher struct {
+	*task.Task
+	db *gorm.DB
+
+	monitor monitoring.Monitor
+
+	input  chan uint64
+	Output chan *Payload
+}
+
+func NewArweaveFetcher(config *config.Config) (self *ArweaveFetcher) {
+	self = new(ArweaveFetcher)
+
+	self.Output = make(chan *Payload)
+
+	self.Task = task.NewTask(config, "arweave-fetcher").
+		WithSubtaskFunc(self.run)
+
+	return
+}
+
+func (self *ArweaveFetcher) WithDB(db *gorm.DB) *ArweaveFetcher {
+	self.db = db
+	return self
+}
+
+func (self *ArweaveFetcher) WithMonitor(monitor monitoring.Monitor) *ArweaveFetcher {
+	self.monitor = monitor
+	return self
+}
+
+func (self *ArweaveFetcher) WithInputChannel(input chan uint64) *ArweaveFetcher {
+	self.input = input
+	return self
+}
+
+func (self *ArweaveFetcher) run() (err error) {
```

Comment:
I would add some sanity check at the end of this function and verify that the newly set sortKey -> lastSortKey sequence is correct...at least for the first few weeks of running this in prod...